python/lsst/daf/butler/queries/_query.py

```diff
@@ -292,6 +298,45 @@ def dimension_records(self, element: str) -> DimensionRecordQueryResults:
         result_spec = DimensionRecordResultSpec(element=self._driver.universe[element])
         return DimensionRecordQueryResults(self._driver, tree, result_spec)
 
+    def general(
+        self,
+        dimensions: DimensionGroup,
+        dimension_fields: Mapping[str, set[str]] = {},
+        dataset_fields: Mapping[str, set[DatasetFieldName]] = {},
```

Comment:
Since we don't need to mutate them, better to make the value types `collections.abc.Set` rather than `set`.

What do you think about making all of these keyword-only arguments, and adding a `*args: str`  variadic argument that we'd filter though `interpret_identifier` (in `queries/_identifiers.py`) and use to update the existing categorized values?

It might also be nice to have a simple way to say "all of the fields necessary for `DatasetRefs` of this dataset type", maybe `datasets={name: ...}` or something?